### DISCUSSION TITLE
How to use rich Traceback with raise errors?

### DISCUSSION THREAD

**Discussion by codenotworking** [Textualize/rich#1461] [Q&A] [answered]
I'm trying to use rich's Traceback with `raise` errors, but the documentation doesn't give an example on it (unless I missed one).  Just for `try/except` blocks.  This is what I have in psuedo-code:
```python
from rich.console import Console

console = Console()

def some_function(something):
    if something == "foo":
        do_something()
    elif something == "bar":
        do_something_else()
    else:
        console.print_exception(show_locals=True)
        raise ValueError("You dun goofed.")
```
I receive the error `ValueError: Value for 'trace' required if not called in except: block`.  I checked the documentation for [rich.Traceback](https://rich.readthedocs.io/en/latest/reference/traceback.html#module-rich.traceback) which states:
`trace (Trace, optional) – A Trace object produced from extract. Defaults to None, which uses the last exception.`.  Unfortunately, I don't know how to proceed.  Some help would be appreciated.

**Comment by willmcgugan** [comment, marked as answer]
The print_exception method will only work when there is an exception to print, i.e. within an `except:` block.

i.e. something like this:

```python
try:
    some_function("foo")
except:
    console.print_exception(show_locals=True)
```
